stl/inc/iterator

```diff
@@ -133,17 +133,22 @@ public:
     using traits_type  = _Traits;
     using istream_type = basic_istream<_Elem, _Traits>;
 
-    constexpr istream_iterator() : _Myistr(nullptr), _Myval() {}
+    static_assert(conjunction_v<is_default_constructible<_Ty>, is_copy_constructible<_Ty>, is_copy_assignable<_Ty>>,
+        "istream_iterator<T> requires T to be default constructible, copy constructible, and copy assignable. (N4835 [istream.iterator]/2)");

```

Comment:
I see a clang-format issue here (this line exceeds 120 columns and must be wrapped). Other than that, looks great. I'll push a clang-format change and port this to the MSVC-internal repo for testing. (For some reason the Azure Pipelines bot isn't responding at the moment.)

Reply:
Hm that is strange, as clang-format was working for me on the other PR. Are we using clang-format 9 now. 

On the other hand I am on a new setup so...

Reply:
Yeah i have clang-format 8 running...

Reply:
Yep, we're using clang-format 9, although its behavior doesn't differ here.

If you're using VSCode, make sure that (1) the file is being opened as C++, and (2) you have Format On Save enabled (or you're just really consistent about Alt+Shift+F). My #263 workspace does both of those things (although you still have to provide a clang-format 9 executable, if the C/C++ Extension's bundled copy is older).

I also personally enable a ruler at 120 columns, but I noticed this one purely by sight.